src/frontend/services/markdown-service.js

```diff
@@ -1,101 +1,108 @@
 export class MarkdownService {
-	static parseMarkdown(text) {
-		const md = require("markdown-it")({
-			html: false,
-			linkify: true,
-		});
+  static parseMarkdown(text) {
+    const md = require("markdown-it")({
+      html: false,
+      linkify: true,
+    });
 
-		const output =
-			text === "" ? "Niente da visualizzare" : md.render(text.trim());
-		return output;
-	}
+    const output =
+      text === "" ? "Niente da visualizzare" : md.render(text.trim());
+    return output;
+  }
 
-	static insertInTextArea(str) {
-		let textarea = MarkdownService.getTextarea();
-		let start = textarea.selectionStart;
-		let finish = textarea.selectionEnd;
+  static insertInTextArea(text) {
+    const editor = MarkdownService.getEditor();
```

Comment:
c'è tantissima duplicazione di codice in questo servizio
prova a rimuoverla e prova a pensare se tutta la logica inserita qui è corretto che sia qui